src/App.test.js

```diff
@@ -0,0 +1,41 @@
+import { screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { renderWithProviders } from "./test/utils"
+import App from './App'
+
+it('renders watch later link', () => {
+  renderWithProviders(<App />)
+  const linkElement = screen.getByText(/watch later/i)
+  expect(linkElement).toBeInTheDocument()
+})
+
+it('search for movies', async () => {
+  renderWithProviders(<App />)
+  await userEvent.type(screen.getByTestId('search-movies'), 'forrest gump')
+  await waitFor(() => {
+    expect(screen.getAllByText('Through the Eyes of Forrest Gump')[0]).toBeInTheDocument()
+  })
+  const viewTrailerBtn = screen.getAllByText('View Trailer')[0]
```

Comment:
I recommend you change this to 
```suggestion
  const viewTrailerBtn = screen.getAllByRole('button', { name: 'View Trailer'})[0]
```
Because you're targeting directly to filter just the buttons and not asking for all the texts 😉